src/Files.App/UserControls/ThemedIcon/ThemedIcon.xaml

```diff
@@ -0,0 +1,343 @@
+										<Setter Target="PART_OutlineIconViewBox.Visibility" Value="Collapsed" />
+										<Setter Target="PART_FilledIconViewBox.Visibility" Value="Collapsed" />
+										<Setter Target="PART_LayeredIconViewBox.Visibility" Value="Visible" />
+									</VisualState.Setters>
+								</VisualState>
+							</VisualStateGroup>
+
+							<VisualStateGroup x:Name="IconColorStates">
+								<VisualState x:Name="Normal">
+									<VisualState.Setters>
+										<Setter Target="PART_OutlinePath.Fill" Value="{ThemeResource ThemedIconBaseBrush}" />
+										<Setter Target="PART_FilledPath.Fill" Value="{ThemeResource ThemedIconAccentBrush}" />
+									</VisualState.Setters>
+								</VisualState>
+								<VisualState x:Name="Critical">
+									<VisualState.Setters>
+										<Setter Target="PART_OutlinePath.Fill" Value="{ThemeResource ThemedIconCriticalBrush}" />
+										<Setter Target="PART_FilledPath.Fill" Value="{ThemeResource ThemedIconCriticalBrush}" />
+									</VisualState.Setters>
+								</VisualState>
+								<VisualState x:Name="Caution">
+									<VisualState.Setters>
+										<Setter Target="PART_OutlinePath.Fill" Value="{ThemeResource ThemedIconCautionBrush}" />
+										<Setter Target="PART_FilledPath.Fill" Value="{ThemeResource ThemedIconCautionBrush}" />
+									</VisualState.Setters>
+								</VisualState>
+								<VisualState x:Name="Success">
+									<VisualState.Setters>
+										<Setter Target="PART_OutlinePath.Fill" Value="{ThemeResource ThemedIconSuccessBrush}" />
+										<Setter Target="PART_FilledPath.Fill" Value="{ThemeResource ThemedIconSuccessBrush}" />
+									</VisualState.Setters>
+								</VisualState>
+								<VisualState x:Name="Neutral">
+									<VisualState.Setters>
+										<Setter Target="PART_OutlinePath.Fill" Value="{ThemeResource ThemedIconNeutralBrush}" />
+										<Setter Target="PART_FilledPath.Fill" Value="{ThemeResource ThemedIconNeutralBrush}" />
+									</VisualState.Setters>
+								</VisualState>
+								<VisualState x:Name="Accent">
+									<VisualState.Setters>
+										<Setter Target="PART_OutlinePath.Fill" Value="{ThemeResource ThemedIconAccentBrush}" />
+										<Setter Target="PART_FilledPath.Fill" Value="{ThemeResource ThemedIconAccentBrush}" />
+									</VisualState.Setters>
+								</VisualState>
+								<VisualState x:Name="Toggle">
+									<VisualState.Setters>
+										<Setter Target="PART_OutlinePath.Fill" Value="{ThemeResource ThemedIconAccentContrastBrush}" />
+										<Setter Target="PART_FilledPath.Fill" Value="{ThemeResource ThemedIconAccentContrastBrush}" />
+									</VisualState.Setters>
+								</VisualState>
+								<VisualState x:Name="HighContrast">
+									<VisualState.Setters>
+										<Setter Target="PART_OutlinePath.Fill" Value="{ThemeResource ThemedIconHighContrastBrush}" />
+										<Setter Target="PART_FilledPath.Fill" Value="{ThemeResource ThemedIconHighContrastBrush}" />
```

Comment:
Why are we doing this manually instead of all brushes just pointing to these two values in high contrast? That would allow us to not having to listening to the high contrast event.

Reply:
We have the option to always use High Contrast icons, for those users who don't like the use of the accent colour.

Reply:
@mdtauk Is this a use case we plan to support? 
@marcelwgn does this affect performance or is there no harm in keeping it just in case?

Reply:
> @mdtauk Is this a use case we plan to support? 
I don't know, you will have to decide on that

> @marcelwgn does this affect performance or is there no harm in keeping it just in case?
It is just a property ``UseContrast`` which works alongside the internal ``_isHighContrast`` bool which is set by the Theme Changed service

Reply:
Those brushes are set when user set Windows high contrast on AND/OR when user make a preference on using high contrast icons (if we decide to support this). If we drop the second one’s support, we should define high contrast brush only in the HighContrast XAML key.

Reply:
I don't know of another way to handle the choice of high contrast, as well as light and dark normal support, without having a high contrast brush and state.

If we drop support for the `UseContrast` property, and only support it with the system theme - then we could then remove the extra brushes and visual states

Reply:
I don't foresee this being an option, but if keeping the code doesn't affect performance or make it harder to maintain, it can't hurt to keep it around.

Reply:
I could differentiate between `UseContrast` and `_isHighContrast` so the HighContrast resources kick in, and the specific High Contrast brush we use is reserved only for the optional use.

Reply:
My 0.02€: If we currently don't support the option to toggle high contrast on for Files regardless of the system settings, I would not go the length to support it now. I agree with @mdtauk that it's not much more to support, however event listeners aren't free either. And the ThemeIcon control might get rendered dozens of times so small costs might add up.

Reply:
They way I see it, HighContrast based on the system rather than the control's "UseContrast" property.  Still requires an event listener, so we can replace the Layered visual state with the outline.

The UseContrast setting basically also replaces the layered icon state with the Outlined icon - however we also have a separate brush resource, which makes the icons use the same brush as the text around it, ensuring it stands out clearer. (our base brush is closer to the TextSecondary brush so they don't feel too intrusive.)

Even if we were to remove the alternate brush choice, and the UseContrast property - and decide never to support it, or custom icon colours for the custom themes in Files - the HighContrast system event is still essential.

Reply:
I am happy to remove the UseContrast additional setting, if no purpose is seen in it.

I know some people use a grey Accent colour or may want to change the icon colour via a custom theme - but if there is no plans to implement that, then so be it.

Reply:
> Those brushes are set when user set Windows high contrast on AND/OR when user make a preference on using high contrast icons (if we decide to support this). If we drop the second one’s support, we should define high contrast brush only in the HighContrast XAML key.

That is the case, in the other repo, I have removed the UseContrast property and states, as well as removed the brushes they used to use.

Reply:
Olay, let's remove and if the need arises, we can refer to this PR and add it back.